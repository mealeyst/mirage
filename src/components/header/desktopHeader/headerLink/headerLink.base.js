import React from 'react'

import { Highlighter, Link } from 'SRC'

export default class BaseHeaderLink extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      hover: false,
      active: false
    }
  }

  enter = () => {
    const { onMouseEnter } = this.props
    this.setState({ hover: true })
    onMouseEnter && onMouseEnter()
  }

  leave = () => {
    const { onMouseLeave } = this.props
    this.setState({ hover: false })
    onMouseLeave && onMouseLeave()
  }

  render () {
    const { className, children, highlightable, highlightOn, ...props} = this.props
    const highlight = this.state[highlightOn]
    delete props.spacing
    return (
      <Link
        {...props}
        className={className}
        onMouseEnter={this.enter}
        onMouseLeave={this.leave}
        light
        underline={false}
        uppercase>
        {highlightable && <Highlighter highlight={highlight}/>}
        <span>{children}</span>
      </Link>
    )
  }
}
