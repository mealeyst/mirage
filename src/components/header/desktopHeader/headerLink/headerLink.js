import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Highlighter } from 'SRC'

import BaseHeaderLink from './headerLink.base'


/** @component */
const HeaderLink = styled(BaseHeaderLink)`
  position: relative;
  ${props => props.spacing ? `margin: 0 1rem;`: ''}
  display: flex;
  cursor: pointer;
  height: 6rem;
  align-items: center;
  text-decoration: none;
  letter-spacing: 0.1rem
  &:active, &:visited {
    color: ${props => props.theme.colors.rocketBlue};
  }
  > span {
    z-index: 1;
    ${props => props.spacing ? `padding: 0.5rem 1.5rem;`: ''}
  }
  .screenReader{
    font-size: 0;
    height: 0;
  }
  ${Highlighter} {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 100%;
    height: auto;
    max-height: 100%;
    max-width: 12rem;
    transform: translateX(-50%) translateY(-50%);
  }
`

HeaderLink.defaultProps = {
  highlightable: true,
  highlightOn: 'hover',
  spacing: true
}

HeaderLink.propTypes = {
  highlightable: PropTypes.bool,
  highlightOn: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.string
  ])
}

export { BaseHeaderLink }
export default HeaderLink
