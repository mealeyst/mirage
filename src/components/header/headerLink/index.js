export { default as HeaderLink } from './headerLink'
