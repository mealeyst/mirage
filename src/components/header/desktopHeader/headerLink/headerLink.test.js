import React from 'react'
import 'jest-styled-components'

import { HeaderLink, Highlighter, Link } from 'SRC'
import { BaseHeaderLink } from './headerLink'

const { shallowWithTheme, mountWithTheme } = global

describe('(Styled Component) HeaderLink', () => {
  const createHeaderLink = (props) => {
    return shallowWithTheme(<HeaderLink {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createHeaderLink())
    .toMatchSnapshot()
  })
})

describe('(Component) BaseHeaderLink', () => {
  const createBaseHeaderLink = (inProps) => {
    const props = {
      highlightable: true,
      highlightOn: 'hover',
      spacing: true,
      ...inProps
    }
    return mountWithTheme(<BaseHeaderLink {...props}/>)
  }

  test('rendering the link text', () => {
    const text = 'Example Link'
    expect(createBaseHeaderLink({children: text}).contains(text)).toBeTruthy()
  })

  test('rendering the Highlighter component by default', () => {
    expect(
      createBaseHeaderLink()
      .find(Highlighter).length
    ).toBeTruthy()
  })

  test('rendering the Highlighter component by default', () => {
    expect(
      createBaseHeaderLink({highlightable: false})
      .find(Highlighter).length
    ).toBeFalsy()
  })
})
