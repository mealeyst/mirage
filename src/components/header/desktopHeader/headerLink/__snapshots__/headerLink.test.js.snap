// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) HeaderLink matching the snapshot 1`] = `
.c0 {
  position: relative;
  margin: 0 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  height: 6rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c0:active,
.c0:visited {
  color: #0073D1;
}

.c0 > span {
  z-index: 1;
  padding: 0.5rem 1.5rem;
}

.c0 .screenReader {
  font-size: 0;
  height: 0;
}

.c0 .c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  width: 100%;
  height: auto;
  max-height: 100%;
  max-width: 12rem;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

<BaseHeaderLink
  className="c0"
  highlightOn="hover"
  highlightable={true}
  spacing={true}
/>
`;
